app/src/org/commcare/android/database/global/models/ApplicationRecord.java

```diff
@@ -56,4 +84,104 @@ public int getStatus() {
     public void setStatus(int status) {
         this.status = status;
     }
+
+    public String getUniqueId() {
+        return uniqueId;
+    }
+
+    public void setUniqueId(String id) {
+        this.uniqueId = id;
+    }
+
+    public String getDisplayName() {
+        return this.displayName;
+    }
+
+    public void setDisplayName(String appName) {
+        this.displayName = appName;
+    }
+
+    public void setArchiveStatus(boolean b) {
+        this.isArchived = b;
+    }
+
+    public boolean isArchived() {
+        return this.isArchived;
+    }
+
+    @Override
+    public String toString() {
+        return this.displayName;
+    }
+
+    public void setResourcesStatus(boolean b) {
+        this.resourcesValidated = b;
+    }
+
+    public boolean resourcesValidated() {
+        return this.resourcesValidated;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord represents an app generated from
+     * a pre-Multiple Apps version of CommCare that does not have profile files
+     * with uniqueId and displayName
+     */
+    public boolean preMultipleAppsProfile() {
+        return this.preMultipleAppsProfile;
+    }
+    
+    public void setPreMultipleAppsProfile(boolean b) {
+        this.preMultipleAppsProfile = b;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord was just generated from the a
+     * different ApplicationRecord format via the db upgrader, because it was 
+     * initially installed on a phone with a pre-Multiple Apps version of CommCare
+     */
+    public boolean convertedByDbUpgrader() {
+        return this.convertedViaDbUpgrader;
+    }
+
+    public void setConvertedByDbUpgrader(boolean b) {
+        this.convertedViaDbUpgrader = b;
+    }
+
+    /**
+     * Completes a full uninstall of the CommCare app that this ApplicationRecord represents
+     */
+    public void uninstall(Context c) {

```

Comment:
I think this method should likely live in a different place. 

The AppRecord/Case/db model classes should try to have a separation of concerns which decouples the data they represent with how it is used, generally.

It looks like a lot of calls in here are stateful from the application (many of the calls are to its static instance), maybe that would be a good place for it?